docs/terms.md

```diff
@@ -0,0 +1,142 @@
+## Suite
+
+A suite of tests.
+A single suite has a directory structure, and many _test spec files_.
+Each test spec file is identified by its path within the suite.
+
+**Example:** `src/suites/*/`
+
+### README
+
+**Example:** `src/suites/**/README.txt`
+
+Describes (in prose) the contents of a subdirectory in a suite.
+
+**Type:** After a README is loaded, it is stored as a `ReadmeFile`.
+
+## IDs
+
+### Test Spec ID
+
+Uniquely identifies a single test spec file.
+Comprised of suite name (`suite`) and test spec file path (`path`).
+
+**Type:** `TestSpecID`
+
+**Example:** `{ suite: 'cts', path: 'command_buffer/compute/basic' }`
+
+### Test Case ID
+
+Uniquely identifies a single test case within a test spec.
+
+Comprised of test name (`test`) and the parameters for a case (`params`).
+(If `params` is null, there is only one case for the test.)
+
+**Type:** `TestCaseID`
+
+**Example:** `{ test: '', params: { 'value': 1 } }`
+
+## Listing
+
+A listing of the **test spec files** in a suite.
+
+This can be generated only in Node, which has filesystem access (see `src/tools/crawl.ts`).
+As part of the build step, a _listing file_ is generated (see `src/tools/gen.ts`) so that the
+test files can be discovered by the web runner (since it does not have filesystem access).
+
+**Type:** `TestSuiteListing`
+
+### Listing File
+
+**Example:**: `out/suites/*/index.js`
+
+## Test Spec
+
+May be either a _test spec file_ or a _filtered test spec_.
+It is identified by a `TestSpecID`.
+
+Always contains one `RunCaseIterable`.
+
+**Type:** `TestSpec`
+
+### Test Spec File
+
+A single `.spec.ts` file. It always contains one _test group_.
+
+**Example:** `src/suites/**/*.spec.ts`
+
+### Filtered Test Spec
+
+A subset of the tests in a single test spec file.
+It is created at runtime, via a filter.
+
+It contains one "virtual" test group, which has type `RunCaseIterable`.
+
+## Test Group / Group
+
+A collection of test cases. There is one per test spec file.
```

Comment:
Is this necessarily true? Is there some use to having more than one group per spec?

Reply:
It's sort of fundamentally true right now because groups are not identifiable within files (i.e. the path to a file uniquely identifies a group).

Reply:
But there's a convention I'm using which is that if you want a hierarchy of tests within a test group, name them with slashes between the parts. I'm planning to actually make this nested in the standalone runner eventually.